docs/src/pages/core-concepts/routing.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Routing
+---
+
+Astro uses **file-based routing** to generate your build URLs based on the file layout of your project `src/pages` directory. When a file is added to the `src/pages` directory of your project, it is automatically available as a route based on its filename.
+
+## Static routes
+
+Astro Components (`.astro`) and Markdown Files (`.md`) are the only currently supported formats for routes. Other file types (like a `.jsx` React component) are not supported as routes in the `src/pages` directory, but you can use any UI component inside of an `.astro` file to achieve a similar result.
+
+The file path within the `src/pages` directory decides the final URL that the page is built to:
+
+```bash
+# Examples
+src/pages/index.astro       -> mysite.com/
+src/pages/about.astro       -> mysite.com/about
+src/pages/about/index.astro -> mysite.com/about
+src/pages/about/me.astro    -> mysite.com/about/me
+src/pages/posts/1.md        -> mysite.com/posts/1
+```
+## Dynamic parameters
+
+Sometimes, you will need to generate many URLs from a single page component. Astro supports this with **dynamic parameters** in the page filename.
+
+Dynamic parameters are encoded into the filename using `[bracket]` notation:
+
+- `pages/blog/[slug].astro` → `/blog/:slug` (`/blog/hello-world`, `/blog/post-2`, etc.)
+- `pages/[username]/settings.astro` → (`/fred/settings`, `/drew/settings`, etc.)
+- `pages/[lang]-[version]/info.astro` → (`/en-v1/info`, `/fr-v2/info`, etc.)
+
+#### Example: Dynamic parameters
+Consider the following page `pages/post/[pid].astro`:
+
+```jsx
+---
+// Example: src/pages/post/[pid].astro
+const {pid} = Astro.request.params;
+---
+<p>Post: {pid}</p>
+```
+
+Any route like `/post/1`, `/post/abc`, etc. will be matched by `pages/post/[pid].astro`. The matched path parameter will be sent as a query parameter to the page.
+
+For example, the route `/post/abc` will have the following `Astro.request.params` object available:
+
+```json
+{ "pid": "abc" }
+```
+
+Multiple dynamic route segments can be combined to work the same way.  The page `pages/post/[pid]/[comment].astro` will match the route `/post/abc/a-comment` and its `query` object will be:
+
+```json
+{ "pid": "abc", "comment": "a-comment" }
+```
+
+## Rest parameters
+
+If you need more flexibility in your URL routing, you can use a rest parameter as a universal catch-all. You do this by adding three dots (`...`) inside your brackets. For example:
+
+- `pages/post/[...slug].astro` → (`/post/a`, `/post/a/b`, `/post/a/b/c`, etc.)
```

Comment:
Very slick

Reply:
TODO: should we throw an error if someone tries something weird in a filename? e.g. is `pages/post/[author]-[...slug]` allowed?